src/Topologies/dss.jl

```diff
@@ -684,7 +677,14 @@ end
 
 Computed unweighted/pure DSS of `data`.
 """
-function dss!(data::DSSDataTypes, topology::Topology2D)
+function dss!(data::DSSTypes1D, topology::IntervalTopology)
+    Nij = DataLayouts.get_Nij(data)
+    length(parent(data)) == 0 && return nothing
```

Comment:
I think there might be more cases here that we can short circuit. What do you think?

Reply:
You mean like whether we are using GL or GLL quadrature? I think we're just assuming GLL quadrature everywhere outside of a few unit tests. Or did you have something else in mind?

Reply:
Just checking `Nh > 1` should be good (+GLL?)

Reply:
We cannot optimize out DSS in a horizontally periodic space when `Nh == 1`, since the opposite sides of the single horizontal element are still collocated. We can only optimize it out when we are using GL instead of GLL, or when there is no data to DSS. To check the first condition, I have defined the function `Quadratures.requires_dss(quadrature_style(space))`. To check the second condition, I am now using `sizeof(eltype(data))`, which is somewhat clearer than the `length(parent(data))` that was originally here.